[Bug]: "Do not use isMounted" linting errors after upgrading to latest `eslint-plugin-react` (v7.36.0)

### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

<!--Brief description-->
Before upgrading from v7.35.2 to v7.36.0, my linting tests were running with no errors, now they throw 4 errors about the usage of isMounted

<!--Show example of your code (as text format), add images/videos/gifs to help explain example-->
`npm run lint` output:
```
  277:5   error    Do not use isMounted                      react/no-is-mounted
  281:21  error    Do not use isMounted                      react/no-is-mounted
  359:5   error    Do not use isMounted                      react/no-is-mounted
  378:5   error    Do not use isMounted                      react/no-is-mounted

✖ 40 problems (4 errors, 36 warnings)
```
<!--and/or Link of repo to where issue is occurring-->


<!--What is happening? / What is the error?-->
We are not using isMounted in our codebase

<!--What command(s) did you run to reproduce issue?-->
full command used: `npm run rm && npm i --dev && npm run build && npm run lint`

I later pinned the version back to v7.35.2 and the errors were no longer present

### Expected Behavior

<!--Brief description-->
Before the upgrade:
```
✖ 36 problems (0 errors, 36 warnings)
```
<!--Show example of code (as text format), add images/videos/gifs to help explain expected behavior-->


### eslint-plugin-react version

v7.36.0

### eslint version

v8.57.0

### node version

v16.20.2

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,11 @@ This change log adheres to standards from [Keep a CHANGELOG](https://keepachange
 
 ## Unreleased
 
+### Fixed
+* [`no-is-mounted`]: fix logic in method name check ([#3821][] @Mathias-S)
+
+[#3821]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3821
+
 ## [7.36.0] - 2024.09.12
 
 ### Added

--- a/lib/rules/no-is-mounted.js
+++ b/lib/rules/no-is-mounted.js
@@ -41,7 +41,8 @@ module.exports = {
         }
         if (
           callee.object.type !== 'ThisExpression'
-          && (!('name' in callee.property) || callee.property.name !== 'isMounted')
+          || !('name' in callee.property)
+          || callee.property.name !== 'isMounted'
         ) {
           return;
         }

--- a/tests/lib/rules/no-is-mounted.js
+++ b/tests/lib/rules/no-is-mounted.js
@@ -57,6 +57,17 @@ ruleTester.run('no-is-mounted', rule, {
         });
       `,
     },
+    {
+      code: `
+        class Hello extends React.Component {
+          notIsMounted() {}
+          render() {
+            this.notIsMounted();
+            return <div>Hello</div>;
+          }
+        };
+      `,
+    },
   ]),
 
   invalid: parsers.all([

